### PR TITLE
Feature/add login page

### PR DESCRIPTION
**Pull request by LennartSchwier**
Add the login page and protected routes for all other components

### CODE REVIEW

File: frontend/src/components/Dashboard.js [hunk 1]
```diff
@@ -27,26 +27,26 @@ export default function Dashboard({latitude, longitude, errorMessage}) {
...
-        history.push("/input");
+    function redirectToTrimInput() {
+        history.push("/triminput");
     }
 }
 
 const PageLayout = styled.div`
 display: grid;
-grid-template-rows: 60px 120px 120px 1fr 60px;
+grid-template-rows: 60px 120px 1fr 60px;
```
Comment (benjamin-ranft) [inline frontend/src/components/Dashboard.js:42]: Try to use more fr or other relative units in grid templates, it'll serve you well for responsiveness in down the line =)

File: frontend/src/components/Input.test.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import Input from "./Input";
+import TrimInput from "./TrimInput";
 import { render } from '@testing-library/react';
 import {MemoryRouter as Router} from "react-router-dom";
 
-describe('component test :: Input', () => {
-    it('renders the Input page and it\'s content', () => {
+describe('component test :: TrimInput', () => {
+    it('renders the TrimInput page and it\'s content', () => {
```
Comment (benjamin-ranft) [inline frontend/src/components/Input.test.js:6]: Tests looking good!

File: frontend/src/service/StormGlassService.js [hunk 3]
```diff
@@ -1,7 +1,15 @@
 import axios from 'axios';
 
+const setHeaders = () => {
+    return {
+        headers: {
+            "Authorization": "Bearer " + localStorage.getItem("jwtToken")
```
Comment (benjamin-ranft) [inline frontend/src/service/StormGlassService.js:6]: Aren't you missing the $-sign here? Mine looks like this " Authorization: `Bearer ${token}`"

**Review by benjamin-ranft — APPROVED** [review]
LGTM

File: frontend/src/routing/ProtectedRoute.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Route , Redirect} from "react-router-dom";
+import jwtDecode from "jwt-decode";
+
+export default function ProtectedRoute(props) {
+
+    const token = localStorage.getItem("jwtToken");
+    const decode = token && jwtDecode(token);
+    console.log(decode);
```
Comment (HennyHaelbich) [inline frontend/src/routing/ProtectedRoute.js:9]: console.log sollte aus dem produktiv Code wieder raus

**Review by Asamya — APPROVED** [review]
LGTM

File: frontend/src/service/StormGlassService.js (hunk 3 above)
Comment (LennartSchwier) [reply frontend/src/service/StormGlassService.js:6]: I think that depends. If you want to use a template literal you need the back-ticks and dollar sign for variables, like you did. Or you can do string concatenation with the plus sign and whatever ticks you prefer.

File: frontend/src/routing/ProtectedRoute.js (hunk 4 above)
Comment (LennartSchwier) [reply frontend/src/routing/ProtectedRoute.js:9]: Du hast vollkommen recht! 😬

**Pull request merged**